llvmlite/tests/test_ir.py

```diff
@@ -166,6 +166,68 @@ def test_function_metadata(self):
         # Check pickling
         self.assert_pickle_correctly(func)
 
+    def test_function_section(self):
+        # Output section
+        func = self.function()
+        func.section = "a_section"
+        asm = self.descr(func).strip()
+        self.assertEqual(asm,
+                         "declare %s section \"a_section\"" %
+                         self.proto)
+        # Check pickling
+        self.assert_pickle_correctly(func)
+
+    def test_function_section_meta(self):
+        # Now with function metadata
+        module = self.module()
+        func = self.function(module)
+        func.section = "a_section"
+        func.set_metadata('dbg', module.add_metadata([]))
+        asm = self.descr(func).strip()
+        self.assertEqual(asm,
+                         """declare i32 @"my_func"(i32 %".1", i32 %".2", double %".3", i32* %".4") section "a_section" !dbg !0"""  # noqa E501
```

Comment:
Given these new tests are concerned with checking information associated with the function declaration and not the form of the prototype itself, could all the assertions be rewritten in a form like:
```python
self.assertEqual(asm, f'declare {self.proto} section "a_section" !dbg !0')
```
i.e. reuse the invariant prototype?

Reply:
Yes, that's a good idea. I have updated the tests.